src/Foundation/NSObject2.cs

```diff
@@ -409,7 +409,9 @@ private void InitializeObject (bool alloced)
 		void CreateManagedRef (bool retain)
 		{
 			HasManagedRef = true;
-			bool isUserType = Runtime.IsUserType (handle);
+			if (!Runtime.TryGetIsUserType (handle, out var isUserType, out var error_message))
+				throw new InvalidOperationException ($"Unable to create a managed reference for the pointer {handle.ToString ()} whose managed type is {GetType ().FullName} because it wasn't possible to get the class of the pointer: {error_message}");
```

Comment:
```suggestion
				throw new InvalidOperationException ($"Unable to create a managed reference for the pointer {handle} whose managed type is {GetType ().FullName} because it wasn't possible to get the class of the pointer: {error_message}");
```
Do we really need the ToString?